### DISCUSSION TITLE
V6: Field Array - After trigger() errors object is empty

### DISCUSSION THREAD

**Discussion by knoefel** [react-hook-form/react-hook-form#4806] [Q&A]
Hi!

I have a nested form where I pass the formMethods (returned by `useForm`) by props. One child of the form is a Field Array. Now I have the following problem: each Field Array item has required fields. If I submit without filling in the required fields, as expected, the errors are added to the errors object. 
If I now delete one of the fields, the parent component (in which I have defined `useForm`) is not re-rendered. Only the child component will be re-rendered. But I need the information about the current error state in the parent component without clicking submit again. Therefore I tried to call the `trigger` method while executing the `remove` method. But then all errors are deleted.

It is easier to understand with an example:

1. open [CodeSandbox](https://codesandbox.io/s/react-hook-form-usefieldarray--trigger-xlxns)
2. open Console
3. add 2 fields to Field Array by pressing "APPEND"
4. press "SUBMIT"
5. `errors` object has 2 errors
6. Remove Field by pressing "DELETE WITH TRIGGER()" .
7. Check console that `errors` object is empty

How can I get the updated error state after deleting a field array entry in the parent component without submitting?

**Comment by bluebill1049** [comment]
perhaps `trigger` at `useEffect`?

**Reply by knoefel** [reply]
Ok, i think this might be a bug. 👻 I tried an older version `react-hook-form@6.13.0`. With this version i could just use the `remove` method and then both Parent and Child will trigger a re-render with the correct remaining `errors`. I don't have to use the `trigger` method at all.

Here is the [CodeSandbox](https://codesandbox.io/s/react-hook-form-usefieldarray--trigger-working-v6-13-0-6rrx4?file=/src/index.js) with this working version. I updated the steps to reproduce in the CodeSandbox. Since `react-hook-form@6.13.1` it doesn't work any more. So it must have to do something with the changes made in that [release](https://github.com/react-hook-form/react-hook-form/releases/tag/v6.13.1).

**Reply by bluebill1049** [reply]
can you try with the latest v6? see if this is resolved?

**Reply by knoefel** [reply]
i checked with `react-hook-form@6.15.6` but the problem remains.
